csharp/BinarySearch.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+ 
+namespace BinarySearch
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            TestBinarySearch();
+        }
+        
+        private static void TestBinarySearch() {
+            int[] a = new int[] { 2, 4, 6, 8, 9 };
+            Console.WriteLine(BinarySearchIterative(a, 9));
```

Comment:
Please use assertions to verify result of algorithm instead of writing it to console.